ykpack/src/types.rs

```diff
@@ -379,19 +379,29 @@ pub enum CallOperand {
     /// A pair: the definition ID and the binary symbol name, if known. If the callee doeesn't have
     /// all of its type parameters instantiated, then there will be no symbol.
     Fn(DefId, Option<String>),
+    /// A dynamic call via a vtable,
+    /// A pair like the Fn variant.
+    Virtual(DefId, Option<String>),
     /// An unknown or unhandled callable.
     Unknown, // FIXME -- Find out what else. Closures jump to mind.
 }
 
+fn sym_name_str(maybe_name: &Option<String>) -> &str {
+    match maybe_name {
```

Comment:
Might be simpler as `maybe_name.map_or_else(|n| n.as_str(), || "<unknown>")`?

Reply:
Ah yeah, true.

Reply:
Given that `"<unknown>"` is a constant I guess you could even do `maybe_name.map_or(|n| n.as_str(), "<unknown>")`.

Reply:
Or maybe not:
```
error[E0515]: cannot return value referencing function parameter `n`
   --> ykpack/src/types.rs:390:48
    |
390 |     maybe_name.map_or_else(|| "<unknown>", |n| n.as_str())
    |                                                -^^^^^^^^^
    |                                                |
    |                                                returns a value referencing data owned by the current function
    |                                                `n` is borrowed here
```

Reply:
Ah, maybe you want the parameter type to be `Option<&str>` rather than `&Option<String>`? [The latter is a bit odd, in general, though there are cases where it might be the right thing.]

Reply:
That works if we use `as_ref()` at the call sites, so I'm happy with that.